src/plans/D3MAavePlan.sol

```diff
@@ -0,0 +1,213 @@
+        emit File(what, data);
+    }
+
+    function file(bytes32 what, address data) external auth {
+        if (what == "tack") tack = InterestRateStrategyLike(data);
+        else revert("D3MAavePlan/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    function _maxBar() internal view returns (uint256) {
+        return tack.getMaxVariableBorrowRate();
+    }
+
+    // --- Automated Rate targeting ---
+    function _calculateTargetSupply(uint256 targetInterestRate, uint256 totalDebt) internal view returns (uint256) {
+        uint256 base = tack.baseVariableBorrowRate();
+        if (targetInterestRate <= base || targetInterestRate > _maxBar()) {
+            return 0;
+        }
+
+        // Do inverse calculation of interestStrategy
+        uint256 variableRateSlope1 = tack.variableRateSlope1();
+        uint256 targetUtil;
+        if (targetInterestRate > base + variableRateSlope1) {
+            // Excess interest rate
+            uint256 r;
+            unchecked {
+                r = targetInterestRate - base - variableRateSlope1;
+            }
+            targetUtil = _rdiv(
+                            _rmul(
+                                tack.EXCESS_UTILIZATION_RATE(),
+                                r
+                            ),
+                            tack.variableRateSlope2()
+                         ) + tack.OPTIMAL_UTILIZATION_RATE();
+        } else {
+            // Optimal interest rate
+            unchecked {
+                targetUtil = _rdiv(
+                                _rmul(
+                                    targetInterestRate - base,
+                                    tack.OPTIMAL_UTILIZATION_RATE()
+                                ),
+                                variableRateSlope1
+                             );
+            }
+        }
+        return _rdiv(totalDebt, targetUtil);
+    }
+
+    function getTargetAssets(uint256 currentAssets) external override view returns (uint256) {
+        uint256 targetInterestRate = bar;
+        if (targetInterestRate == 0) return 0;  // De-activated
+
+        uint256 totalDebt = stableDebt.totalSupply() + variableDebt.totalSupply();
+
+        uint256 totalPoolSize = dai.balanceOf(adai) + totalDebt;
+
+        uint256 targetTotalPoolSize = _calculateTargetSupply(targetInterestRate, totalDebt);
+        if (targetTotalPoolSize >= totalPoolSize) {
+            // Increase debt (or same)
+            return currentAssets + (targetTotalPoolSize - totalPoolSize);
+        } else {
+            // Decrease debt
+            unchecked {
+                uint256 decrease = totalPoolSize - targetTotalPoolSize;
+                if (currentAssets >= decrease) {
+                    return currentAssets - decrease;
+                } else {
+                    return 0;
+                }
+            }
+        }
+    }
+
+    function active() public view override returns (bool) {
```

Comment:
as mentioned to @iamchrissmith in chat, I noticed that Aave is using the [pause](https://docs.aave.com/developers/v/2.0/the-core-protocol/lendingpool#paused) breaker pattern in the lending pool (with the `whenNotPaused` modifier in [deposit](https://github.com/aave/protocol-v2/blob/master/contracts/protocol/lendingpool/LendingPool.sol#L109), [withdraw](https://github.com/aave/protocol-v2/blob/master/contracts/protocol/lendingpool/LendingPool.sol#L146), ...), that might be taken into consideration in checking the `active` state here or in the `pool`, considering reintroducing the pool `active` view function back and targeting `pause` eventually.

Reply:
Sorry but don't understand what is this.

Reply:
one thing we could do is have a check in `maxWithdraw` and `maxDeposit` like this:

```solidity
if (pool.paused) return 0
```

That way we wouldn't try to deposit/withdraw when it is paused.  `!active` will put us in an unwind state and we won't be able to unwind if their protocol is paused

Reply:
> Sorry but don't understand what is this.

basically @naszam found while reviewing the "remove `active` from pool" PR that aave can pause their pool to prevent deposit/withdraw which you were the other approver on

Reply:
Right, but if also withdraw is paused, then there is not point to check this, the whole thing will revert anyway.

Reply:
right, that is why i suggested the `maxWithdraw`/`maxDeposit` since that would effectively pause us without a revert.  it is the only way i see of incorporating their pause into our code without going back down the pause path ourselves (which i don't want to do). assuming we care to integrate their pause into our code beyond the revert effect

Reply:
I don't think so unless @naszam feels strongly that we need to add this to the `maxWithdraw`/`maxDeposit` functions in the aave pool

Reply:
I'm going to resolve it for now as there seems not to be consensus around checking `paused` state in the Aave lending pool.
I'll dig more into that, run some tests to validate some assumptions and open an issue or address it in a separate pr eventually.